src/test/java/org/broadinstitute/hellbender/tools/spark/sv/FindBreakpointEvidenceSparkUnitTest.java

```diff
@@ -116,38 +117,40 @@ public void generateFastqsTest() {
         expectedAssemblyQNames.stream()
                 .map(qName -> new FindBreakpointEvidenceSpark.QNameAndInterval(qName, 0))
                 .forEach(qNameMultiMap::add);
-        final String expectedFile = fastqFile; // this is NOT redundant -- local copy necessary for lambda serialization
-        FindBreakpointEvidenceSpark.generateFastqs(ctx, qNameMultiMap, reads, 2, true,
-                intervalAndFastqBytes -> compareFastqs(intervalAndFastqBytes, expectedFile));
+        FindBreakpointEvidenceSpark.generateFastqs(ctx,qNameMultiMap,reads,2,true,true,new FastqComparator(fastqFile));
     }
 
-    private static Tuple2<Integer, String> compareFastqs(
-            final Tuple2<Integer, List<byte[]>> intervalAndFastqBytes,
-            final String fastqFile ) {
-        final List<byte[]> fastqList = intervalAndFastqBytes._2;
-        SVFastqUtils.sortFastqRecords(fastqList);
-        final byte[] concatenatedFastqs = new byte[fastqList.stream().mapToInt(fastq -> fastq.length).sum()];
-        int idx = 0;
-        for ( final byte[] fastq : fastqList ) {
-            System.arraycopy(fastq, 0, concatenatedFastqs, idx, fastq.length);
-            idx += fastq.length;
-        }
-
-        final String expectedFile = fastqFile+intervalAndFastqBytes._1();
-        final ByteArrayInputStream actualStream = new ByteArrayInputStream(concatenatedFastqs);
-        try( InputStream expectedStream = new BufferedInputStream(new FileInputStream(expectedFile)) ) {
-            int val;
-            while ( (val = expectedStream.read()) != -1 )
-                Assert.assertEquals(actualStream.read(), val);
-            Assert.assertEquals(actualStream.read(), -1);
-        }
-        catch ( final IOException ioe ) {
-            throw new GATKException("can't read expected values", ioe);
+    private static final class FastqComparator implements FindBreakpointEvidenceSpark.FASTQHandler {
```

Comment:
I believe this "complicated" code is indicating that the FASTQHandler interface may deserve a 1st class citizen treatment, so that if this test tries to access the `FASTQHandler` in `FindBreakpointEvidenceSpark`, the checks wouldn't complain.